### ISSUE TITLE
Add like/dislike system

### ISSUE THREAD

**Issue by lavelinevgeny** [BookStackApp/BookStack#4722]
### Describe the feature you'd like

Adding like (thumbs up) and unlike (thumbs down) functionality for books/pages/chapter

### Describe the benefits this would bring to existing BookStack users

If you have already liked a post/page and now you dislike it, then the old voting will be cancelled and vice-versa. You can reset the settings to default and the like/unlike counts for all/selected posts/pages as well. It comes with two widgets, one to display the most liked posts/pages for a given time range and another to show recently liked posts

### Can the goal of this request already be achieved via other means?

no

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

### How long have you been using BookStack?

Under 3 months

**Comment by ssddanbrown** [comment]
Thanks for the suggestion @lavelinevgeny.
This is along the same lines as previous issues #1524 and #505. Within those, I've closed them as out of scope, which would be my position here also.

Happy to hear a case for such a feature, but I'd need to understand the benefits this would bring to existing BookStack users. I know you filled this part of the issue form out, but you explained your desired technical implementation, rather than how this would fundamentally benefit existing BookStack users.

**Comment by lavelinevgeny** [comment]
@ssddanbrown, sorry, I only looked at similar tasks among open tasks.

I'll try to develop my idea.

I think that the like system will give site users an important characteristic for a specific article/instruction, when it will be clear how much they can trust the information on the current page.

This will be useful even without additional widgets.

If it is possible to display pages with negative reviews (when there are more negative reviews than positive ones), this will give a signal to the author to improve such pages.

You can implement this function optionally so that only those who need it can enable it.

**Comment by ClazOS** [comment]
I agree that this has some nice potential use cases: allowing crowd-sourcing data collection could allow an Editor to gauge which content was meeting a minimum standard of quality or needed attention from the editing users. Whether on an internal or public forum these voting systems are popular because they work: most people aren't inclined toward misusing a feature like this, and there is a real benefit to the editor getting that feedback to know where to focus attention. And yes, there are certainly additional emergent features that might come out of this, but a basic tally creating a hierarchy of quality for "essential" pages would definitely be something I would use on my set up

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/4722
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
